package/src/languageService/kustoLanguageService.ts

```diff
@@ -1058,6 +1058,7 @@ class KustoLanguageService implements LanguageService {
         this._kustoJsSchemaV2 = this._kustoJsSchemaV2.WithParameters(
             KustoLanguageService.toBridgeList([...scalarSymbols, ...tabularSymbols])
         );
+        this._script = this._script.WithGlobals(this._kustoJsSchemaV2);
```

Comment:
I noticed that other `set*()` methods were setting `this._script`, so I updated it using a method that sounded sensible. Not at all sure if this is actually a good idea.

Reply:
The script (CodeScript) is the abstraction of the editor document that breaks the text down into multiple separate query blocks and offers intellisense API's (CodeService) for each block separately. The script (and the blocks/code-services) hold onto the global state that the individual intellisense services will use.  So, if you update anything stored by the global state, you need to update the script with that change for those services to have the correct info.

Reply:
Ok, so it sounds like this is fine? @noam-shapira Do you know how changes to this._script propagate to the editor? I don't see a `triggerUpdates()` call or anything similar.

Reply:
Don't know what trigger the editor recalculation of providers data (validation, coloring, etc).
I don't see any trigger after setSchema